packages/astro/src/@types/astro.ts

```diff
@@ -1096,12 +1087,111 @@ export interface AstroAdapter {
 
 type Body = string;
 
-export interface APIContext {
+// Shared types between `Astro` global and API context object
+interface AstroSharedContext<Props extends Record<string, any> = Record<string, any>> {
+	/**
+	 * The address (usually IP address) of the user. Used with SSR only.
+	 */
+	clientAddress: string;
+	/**
+	 * Utility for getting and setting cookies values.
+	 */
 	cookies: AstroCookies;
-	params: Params;
+	/**
+	 * Information about the current request. This is a standard [Request](https://developer.mozilla.org/en-US/docs/Web/API/Request) object
+	 */
 	request: Request;
+	/**
+	 * A full URL object of the request URL.
+	 */
+	url: URL;
+	/**
+	 * Parameters passed to a dynamic page generated.
+	 */
+	params: Params;
+	/**
+	 * List of props passed.
+	 */
+	props: Props;
+	/**
+	 * Redirect to another page (**SSR Only**)
+	 */
+	redirect(path: string, status?: number): Response;
+}
+
+export interface APIContext<Props extends Record<string, any> = Record<string, any>>
+	extends AstroSharedContext<Props> {
+	site: URL | undefined;
+	generator: string;
+	/**
+	 * A full URL object of the request URL.
+	 * Equivalent to: `new URL(request.url)`
+	 */
+	url: AstroSharedContext['url'];
+	/**
+	 * Parameters passed to a dynamic page generated using `getStaticPaths`.
+	 *
+	 * Example usage:
+	 * ```ts
+	 * export function getStaticPaths() {
+	 *   return [
+	 *     { params: { id: '0' }, props: { name: 'Sarah' } },
+	 *     { params: { id: '1' }, props: { name: 'Chris' } },
+	 *     { params: { id: '2' }, props: { name: 'Fuzzy' } },
+	 *   ];
+	 * }
+	 *
+	 * export async function getStaticProps({ params }) {
+	 *  return {
+	 * 	  body: `Hello user ${params.id}!`,
+	 *  }
+	 * }
+	 * ```
+	 *
+	 * [context reference](https://docs.astro.build/en/guides/api-reference/#contextparams)
+	 */
+	params: AstroSharedContext['params'];
+	/**
+	 * List of props passed from `getStaticPaths`.
```

Comment:
As above: do we need to specify this limitation?

```suggestion
	 * List of props passed from `getStaticPaths`. Only available to static builds.
```